ballerina/modules/parser/operation_node.bal

```diff
@@ -69,4 +73,22 @@ public class OperationNode {
     public isolated function getSelections() returns Selection[] {
         return self.selections;
     }
+
+    public isolated function addVariableDefinition(VariableDefinition varDef) {
+        if (self.variables.hasKey(varDef.name)) {
+            string message = string`There can be only one variable named "$${varDef.name}"`;
+            Location location = <Location> varDef?.location;
+            self.errors.push({message: message, locations:[location]});
+        } else { 
+            self.variables[varDef.name] = varDef;
+        }
+    }
+
+    public isolated function getVaribleDefinitions() returns map<VariableDefinition> {
+        return self.variables;
+    }
```

Comment:
Since we need one variable at a time, a better approach would be creating a `hasVariable` function and a `getVariable` function. This way, we can check whether a variable is defined and if it is defined get only that variable when needed. Otherwise, we have to get the map all time and process it every time.

Reply:
If so,  to access the above functions, we have to pass the ```OperationNode``` up to ```visitField()``` or get the operation node from ```DocumentNode```. Currently,we are passing ```map<VariableDefinition>``` of ```OperationNode``` up to there. WDYT?

Reply:
Sounds right. Let's keep it as it iss then.